builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -81,6 +81,15 @@ func resourceCloudStackInstance() *schema.Resource {
 				Set:      schema.HashString,
 			},
 
+			"affinity_group_names": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
```

Comment:
Just checked and this is not needed: http://cloudstack.apache.org/docs/api/apidocs-4.8/root_admin/updateVMAffinityGroup.html So we can lose the `ForceNew`